vowpalwabbit/search.cc

```diff
@@ -1376,17 +1373,32 @@ void generate_training_example(search_private& priv, polylabel& losses, float we
         lab.costs[0].x = losses.cs.costs[a-start_K].x;
         //cerr << "cost[" << a << "] = " << losses[a] << " - " << min_loss << " = " << lab.costs[0].x << endl;
         ec.in_use = true;
-        uint64_t old_offset = ec.ft_offset;
+        
+        // store the offset to restore it later
+        tmp_offsets.push_back(ec.ft_offset);
```

Comment:
There is a vectorization of things that previously were not vectors here that I don't understand yet.